src/rgw/rgw_op.cc

```diff
@@ -5279,29 +5279,38 @@ void RGWRestoreObj::execute(optional_yield y)
     ldpp_dout(this, 1) << "failed to fetch get_obj_attrs op ret = " << op_ret << dendl;
     return;
   }
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_MANIFEST);
+  attrs = s->object->get_attrs();
+  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+  auto expire_iter = attrs.find(RGW_ATTR_RESTORE_EXPIRY_DATE);
+  auto type_iter = attrs.find(RGW_ATTR_RESTORE_TYPE);
   if (attr_iter != attrs.end()) {
-    RGWObjManifest m;
-    decode(m, attr_iter->second);
-    RGWObjTier tier_config;
-    m.get_tier_config(&tier_config);
-    if (m.get_tier_type() == "cloud-s3") {
-      ldpp_dout(this, 20) << "execute: expiry days" << expiry_days <<dendl;
-      op_ret = handle_cloudtier_obj(s, this, driver, attrs, false, expiry_days, true, y);
-      if (op_ret < 0) {
-        ldpp_dout(this, 4) << "Cannot get cloud tiered object: " << *s->object
-        <<". Failing with " << op_ret << dendl;
-        if (op_ret == -ERR_INVALID_OBJECT_STATE) {
-          s->err.message = "This object was transitioned to cloud-s3";
-        }
-      }
-    } else {
-      ldpp_dout(this, 20) << "not cloud tier object erroring" << dendl;
-      op_ret = -ERR_INVALID_OBJECT_STATE;
+    bufferlist bl = attr_iter->second;
+    auto iter = bl.cbegin();
+    decode(restore_status, iter);
+  }
+
+  if (expire_iter != attrs.end()) {
+    bufferlist bl = expire_iter->second;
+    auto iter = bl.cbegin();
+    decode(expiration_date, iter);
+  }
+
+  if (type_iter != attrs.end()) {
+    
+    bufferlist bl = type_iter->second;
+    auto iter = bl.cbegin();
+    decode(rt, iter);
+  }
+  if (attr_iter != attrs.end() || restore_status == rgw::sal::RGWRestoreStatus::None) {
```

Comment:
no need IMHO

Reply:
if it's not there then for the first request it takes RestoreAlreadyInprogress status and throws 409.